File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        result = []
        count = {}
        for item in nums:
            if item in count:
                count[item] += 1
            else:
                count[item] = 1

        buckets = [[] for _ in range(len(nums) + 1)]

        for num, freq in count.items():
            buckets[freq].append(num)

        results = []
        for freq in range(len(buckets) - 1, 0, -1):
            for num in buckets[freq]:
                results.append(num)
            if len(results) == k:
                return results

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    removeDuplicates(nums: number[]): number {
        let k = 1
        for (let i = 1; i < nums.length; i++) {
            if (nums[i - 1] !== nums[i]) {
                nums[k++] = nums[i] 
            }
        }
        return k
    }
}
